File: go-code/organizing_data/replace_derived_variable_with_query/after/order.go
package after

type Order struct {
	customer           Customer
	customerRepository CustomerRepository
}

// STEP 2: Call customer from the repository when creating an order
func NewOrder(customerRepository CustomerRepository, data interface{}) *Order {
	return &Order{
		customerRepository: customerRepository,
		customer:           customerRepository.RegisterCustomer(""), //Id should come from the data object
	}
}

File: java-code/src/main/java/refactoring/organizing_data/change_value_to_reference/after/Order.java
package refactoring.organizing_data.change_value_to_reference.after;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
public class Order {
  private final Customer customer;
  private final CustomerRepository customerRegistry;

  public Order(CustomerRepository customerRegistry, Object data) {
    this.customerRegistry = customerRegistry;
    String id = null; //It comes from data objet
    this.customer = customerRegistry.registerCustomer(id);
  }
}
